src/sdk/nsfs_glacier_backend/tapecloud.js

```diff
@@ -396,6 +396,13 @@ class TapeCloudGlacierBackend extends GlacierBackend {
 
             await fh.replacexattr(fs_context, undefined, GlacierBackend.XATTR_RESTORE_REQUEST);
         } catch (error) {
+            // restore does check if a file exist or not before triggering eeadm call but the file could get deleted
+            // after `restore` performs that check so check it here once again
+            if (error.code === 'ENOENT') {
+                dbg.warn(`TapeCloudGlacierBackend._finalize_restore: log entry unexpectedly not found: ${entry.path}`);
+                return;
+            }
```

Comment:
it would be better if this was in a separate try-catch which is only on the open() call, to make sure we don't add any other fs calls in the flow that would be interpreted as missing file accidentally

Reply:
Updated the PR